README.md

```diff
@@ -1,63 +1,163 @@
+4. Run `make iam` to provision keycloak on the created application with terraform.
+5. Run `make clear-deploy` to clear pre-existing packages in the app and upload current NPL and migration sources.
+
+#### Option 2: Using the maven plugin
 
-## Building, running python & webapp locally, NPL on PaaS
+1. Edit run configurations by clicking on the three vertical dots icon at the top right of IntelliJ
+2. Add a new configuration by clicking on the `+` icon and selecting `Deploy to Noumena Cloud`
+3. Input the following values:
+    - `Server base URL`: https://portal.noumena.cloud
+    - `Application ID`: The application ID found on the settings page of your app in the Noumena Cloud UI
+    - `Username`: Your email address for Noumena Cloud
+    - `Password`: Your password for Noumena Cloud
+    - `Source path`: Use the file navigator to navigate to `./npl/src/main` inside the project
 
-### NPL and keyloack
+4. Click `Run` to save and run the configuration
 
-1. Add the following variables to your shell
-```
-export NC_BASE_URL=https://portal.noumena.cloud
-export NC_EMAIL=your.email@your.domain
-export NC_ENV=PROD
-```
-2. Run `make create-app` to create the application with name defined in Makefile.
-3. Run `make iam` to provision keycloak on the created PaaS application with terraform.
-4. Run `make clear-deploy` to clear pre-existing packages in the app and upload current NPL and migration sources.
+In addition, Keycloak can be configured from the `Services` tab in the Noumena Cloud UI.
 
-### Python
+#### Option 3: Using the Noumena Cloud UI
 
-The './python' folder contains a python service interacting with the configured engine.
+1. Create a zip file by running the `zip -r sources.zip npl/src/main/` command in the root directory.
+2. In the Noumena Cloud UI, click on `Upload packages` and upload the zip file.
+
+### Python-Listener
+
+The `./python-listener` folder contains a python service interacting with the configured engine.
+In this setup, the python listener service runs locally and connects to the engine on Noumena Cloud.
 
 #### Setup
 
-1. set up & activate your Python venv
-2. run `mvn install` to generate the python client
-3. run `cd python && pip install -r requirements.txt` 
+1. In the root directory, set up & activate your Python venv
+2. run `mvn install` to generate the python client from the NPL code
+3. run `cd python-listener && pip install -r requirements.txt`
 
 Note: it is needed to install requirements every time NPL code is changed.
 This is because the generated code is installed as a package for easier use.
 
 #### Running
 
-run `cd python && python main.py`
-
-and in another terminal
-
-run `cd python && python notification.py`
+From the root directory, run `cd python-listener && python main.py`
 
 ### Webapp
 
-The './webapp' folder contains a typescript frontend service interacting with the configured engine.
+The `./webapp` folder contains a typescript frontend service interacting with the configured engine.
```

Comment:
```suggestion
The `./webapp` folder contains a Typescript frontend service for interacting with the configured Engine.
```

Reply:
I guess in general just make sure `Python`, `Engine`, and `Typescipt` are proper nouns everywhere